great_expectations/core/factory/suite_factory.py

```diff
@@ -103,9 +104,24 @@ def get(self, name: str) -> ExpectationSuite:
     def all(self) -> Iterable[ExpectationSuite]:
         """Get all ExpectationSuites."""
         dicts = self._store.get_all()
-        # Even though we deserialize here, all the validation has already
-        # been done via marshmallow in the get_all() call
-        return [self._deserialize(suite_dict) for suite_dict in dicts]
+        # Marshmallow validation was done in the previous get_all() call for
```

Comment:
This PR started with just adding posthog events but I noticed we could still die with a pydantic error in the suite factory so I added this fix.